### PR TITLE
get rid of some casts

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: libmscore/beam.cpp [hunk 1]
```diff
@@ -2392,7 +2392,7 @@ bool Beam::setProperty(P_ID propertyId, const QVariant& v)
 QVariant Beam::propertyDefault(P_ID id) const
       {
       switch (id) {
-            case P_ID::STEM_DIRECTION: return Direction(Direction::AUTO);
+            case P_ID::STEM_DIRECTION: return Direction::AUTO;
```
Comment (wschweer) [inline libmscore/beam.cpp]: this is wrong, Direction::AUTO has not type Direction and must be casted. Direction is automatically converted to QVariant.

File: libmscore/chord.cpp [hunk 2]
```diff
@@ -2897,7 +2897,7 @@ QPointF Chord::layoutArticulation(Articulation* a)
 
 void Chord::reset()
       {
-      score()->undoChangeProperty(this, P_ID::STEM_DIRECTION, Direction(Direction::AUTO));
+      score()->undoChangeProperty(this, P_ID::STEM_DIRECTION, Direction::AUTO);
```
Comment (wschweer) [inline libmscore/chord.cpp]: same here

**Comment by lasconic** [comment]
No cast in assignment to a Direction is fine, it will automatically converted. But here, we need the right type to be converted to QVariant.

File: libmscore/chord.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [inline libmscore/chord.cpp]: How is this different from 8aff920 ?

Comment (wschweer) [inline libmscore/chord.cpp]: We need a QVariant of type Direction and not int. Direction is automatically converted to QVariant so QVariant a = Direction(Direction::AUTO) is ok, QVariant a = Direction::AUTO; not. QVariant a = int(Direction...) gives the wrong (int) QVariant.
Somehow i could not manage to get Direction::AUTO converted automatically to QVariant of type Direction. Its ambiguous somehow.

Comment (Jojo-Schmitz) [inline libmscore/chord.cpp]: so `undoChangeProperty(P_ID::STEM_DIRECTION, ds->stemDirection(pitch));` would be OK, but `undoChangeProperty(this, P_ID::STEM_DIRECTION, Direction::AUTO);` is not?

Comment (Jojo-Schmitz) [inline libmscore/chord.cpp]: And why does it work in  8aff920?

Comment (wschweer) [inline libmscore/chord.cpp]: oops, https://github.com/musescore/MuseScore/commit/8aff920cfd84c8ead4834874da3b0fb2732fc3d7 is wrong. It must be read Direction(Direction::AUTO), sorry for the confusion.
It seems to be an example of why automatic type conversions are bad.

Comment (Jojo-Schmitz) [inline libmscore/chord.cpp]: Thanks for explaining

**Pull request closed without merging**